packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -86,6 +93,9 @@ export class WaferMap extends FoundationElement {
             this.canvasSideLength = Math.min(height, width);
         });
         this.resizeObserver.observe(this);
+        this.canvas.addEventListener('wheel', event => event.preventDefault(), {
```

Comment:
We should remove this event listener in the `disconnectedCallback`, otherwise we'll end up with duplicate listeners if the same element is added, removed, and re-added to the page.

Reply:
@arinluca333 I don't think the approach you used in `removeEventListener()` will work. I believe the add/remove functions track the listener function by reference. You're attempting to remove a new listener function which is a copy of the original one. So removeEventListener won't find any listener that matches it and will silently fail.

From the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener):
>Calling removeEventListener() with arguments that do not identify any currently registered [event listener](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#the_event_listener_callback) on the EventTarget has no effect.

Instead you need to save the event listener to a class member variable and pass that variable to both `addEventListener` and `removeEventListener`.

Reply:
I filed #964 to you to track fixing this.

Reply:
Yep, the exact same function reference must be used, concretely (pseudocode):

```
class {
    private wheelHandler = () => {};

    connectedCallback() { this.canvas.addEventListener('wheel', wheelHandler);}
    disconnectedCallback() { this.canvas.removeEventListener('wheel', wheelHandler); }
}
```